language/diem-framework/modules/0L/EpochBoundary.move

```diff
@@ -146,13 +147,35 @@ module EpochBoundary {
         };
 
         // If the cardinality of validator_set in the next epoch is less than 4, 
-        // we keep the same validator set. 
-        if (Vector::length<address>(&proposed_set) <= 3) proposed_set = *&top_accounts;
+        // we keep the same validator set, removing validators who have removed themselves 
+        if (Vector::length<address>(&proposed_set) <= 3) {
+            let eligible_validators = ValidatorUniverse::get_eligible_validators(vm);
+            proposed_set = DiemSystem::get_val_set_addr();
```

Comment:
Could you elaborate in the comment the detailed  intent? It's not entirely clear to me when for example a validator would be in the `eligible_validators` (validator that submitted a proof in the last epoch) but not in the `current_validator_set` (all the validators from the last epoch).  I'm probably being dense, but a bit more explanation would be helpful!  e.g.
"if we do not have enough validators in the proposed_set, recalculate the proposed_set, ignoring the `top_accounts` and jailed accounts, and take all the current validator set for the epoch `DiemSystem::get_val_set_addr()` and the validators who "submitted a mined proof correctly in the current epoch".